### PR TITLE
fix(condo): DOMA-9647 pushing newItem in all scopes

### PR DESCRIPTION
**Pull request by tolmachev21**
When sending news, not all residents from the scope received a notification (there was a limit of 100 scopes). Now any number of scopes will be supported.

1. A test was written that checks that 101 residents receive notification of the news.
2. The task has been changed, which notifies the resident about the news in which the limit of 100 scopes has been lifted

### CODE REVIEW

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js [hunk 1]
```diff
@@ -60,7 +61,7 @@ async function sendNotifications (context, newsItem, taskId) {
 
     checkSendingPossibility(newsItem)
 
-    const scopes = await NewsItemScope.getAll(context, { newsItem: { id: newsItem.id } })
+    const scopes = await find('NewsItemScope', { newsItem: { id: newsItem.id } })
```
Comment (SavelevMatthew) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js:64]: `find` and other direct utils (`getById`, `getByCondition`) and others does not filter out deleted objects by default. You must add it by yourself

Comment (SavelevMatthew) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js:64]: Something like this:
```suggestion
    const scopes = await find('NewsItemScope', { newsItem: { id: newsItem.id, deleledAt: null }, deleledAt: null })
```

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 2]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                unitType: unitType1,
+                unitName: unitName1,
+            })
+
+            // News item for particular unit
+            const [newsItem1] = await createTestNewsItem(
+                adminClient,
+                o10n,
+                {
+                    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
```
Comment (SavelevMatthew) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:166]: Use faker for that

Comment (SavelevMatthew) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:166]: https://v7.fakerjs.dev/api/

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 3]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                        transport: 'push',
+
+                        // ADR-7 way check
+                        transportsMeta: [expect.objectContaining({
+                            transport: 'push',
+                        })],
+                    }),
+                    meta: expect.objectContaining({
+                        title: 'Lorem ipsum dolor sit amet, consectetur...',
+                        body: 'Commodo viverra maecenas accumsan lacus vel facilisis volutpat est velit. Et malesuada fames ac turpis egestas sed tempus urna et. At augue eget...',
```
Comment (SavelevMatthew) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:225]: Why body differs from payload above? If you test crop logic, test it programmatically, not hard-coded

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 4]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+            const [newsItem1] = await createTestNewsItem(
+                adminClient,
+                o10n,
+                {
+                    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+                    body: 'Commodo viverra maecenas accumsan lacus vel facilisis volutpat est velit. Et malesuada fames ac turpis egestas sed tempus urna et. At augue eget arcu dictum varius duis at. Tempus quam pellentesque nec nam aliquam sem et tortor consequat. Enim sit amet venenatis urna cursus eget nunc scelerisque viverra. Urna cursus eget nunc scelerisque viverra. Ornare aenean euismod elementum nisi quis eleifend quam. Quis hendrerit dolor magna eget est. Gravida cum sociis natoque penatibus et.',
+                }
+            )
+
+            // create a many newsItemScopes without Residents
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:171]:
```suggestion
            // create NewsItemScopes without Residents 
```

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 5]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+            // create a many newsItemScopes without Residents 
+            for (let i = 0; i <= 100; i++) {
+                await createTestNewsItemScope(adminClient, newsItem1, {
+                    property: { connect: { id: property.id } },
+                    unitType: FLAT_UNIT_TYPE,
+                    unitName: faker.lorem.word(),
+                })
+            }
+
+            // create last newsItemScope with created Resident
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:180]:
```suggestion
            // create NewsItemScope with Resident
```

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 6]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                devicePlatform: DEVICE_PLATFORM_ANDROID,
+                appId: APP_RESIDENT_ID_ANDROID,
+                pushToken: getRandomFakeSuccessToken(),
+            })
+
+            await syncRemoteClientByTestClient(residentClient1, payload)
+
+            const messageWhere = { user: { id: residentClient1.user.id }, type: NEWS_ITEM_COMMON_MESSAGE_TYPE }
+
+            // Publish news item to make it send-able
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:197]:
```suggestion
            // Publish NewsItem to make it sendable
```

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 7]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                        // old way check
+                        transport: 'push',
+
+                        // ADR-7 way check
+                        transportsMeta: [expect.objectContaining({
+                            transport: 'push',
+                        })],
+                    }),
+                    meta: expect.objectContaining({
+                        title: 'Lorem ipsum dolor sit amet, consectetur...',
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:224]: ^^

**Review by toplenboren — CHANGES_REQUESTED** [review]

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 8]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                adminClient,
+                o10n,
+                {
+                    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+                    body: 'Commodo viverra maecenas accumsan lacus vel facilisis volutpat est velit. Et malesuada fames ac turpis egestas sed tempus urna et. At augue eget arcu dictum varius duis at. Tempus quam pellentesque nec nam aliquam sem et tortor consequat. Enim sit amet venenatis urna cursus eget nunc scelerisque viverra. Urna cursus eget nunc scelerisque viverra. Ornare aenean euismod elementum nisi quis eleifend quam. Quis hendrerit dolor magna eget est. Gravida cum sociis natoque penatibus et.',
+                }
+            )
+
+            // create a many newsItemScopes without Residents 
+            for (let i = 0; i <= 100; i++) {
```
Comment (Alllex202) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:172]: better to write a comment why you need to create 100 scopes here

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 9]
```diff
@@ -143,6 +145,98 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+                expect(message1).toBeDefined()
+                expect(message1.id).toMatch(UUID_RE)
+
+                expect(message1).toEqual(expect.objectContaining({
+                    status: MESSAGE_SENT_STATUS,
+                    processingMeta: expect.objectContaining({
+                        // old way check
+                        transport: 'push',
+
+                        // ADR-7 way check
```
[the remark this answers is not in the document]
Comment (tolmachev21) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:218]: But at the moment ADR-7 is the latest version for sending notifications

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js [hunk 10]
```diff
@@ -170,7 +171,7 @@ async function notifyResidentsAboutNewsItem (newsItemId) {
 
         const newsItem = await NewsItem.getOne(context, { id: newsItemId })
 
-        await sendNotifications(context, newsItem, taskId)
+        await sendNotifications(newsItem, taskId)
```
Comment (Alllex202) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js:174]: Why did you remove the context here?

**Review by Alllex202 — CHANGES_REQUESTED** [review]

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 11]
```diff
@@ -24,6 +25,8 @@ const { createTestProperty } = require('@condo/domains/property/utils/testSchema
 const { createTestResident } = require('@condo/domains/resident/utils/testSchema')
 const { makeClientWithResidentUser } = require('@condo/domains/user/utils/testSchema')
 
+const TITLE_MAX_LEN = 50
+const BODY_MAX_LEN = 150
```
Comment (Alllex202) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:29]: I think it is not worth duplicating these constants, it is better to import them

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 12]
```diff
@@ -99,8 +107,8 @@ describe('notifyResidentsAboutNewsItem', () => {
                         })],
                     }),
                     meta: expect.objectContaining({
-                        title: 'Lorem ipsum dolor sit amet, consectetur...',
-                        body: 'Commodo viverra maecenas accumsan lacus vel facilisis volutpat est velit. Et malesuada fames ac turpis egestas sed tempus urna et. At augue eget...',
+                        title: truncate(newsItemTitle, { length: TITLE_MAX_LEN, separator: ' ', omission: '...' }),
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:110]: is truncate really not necessary here?

Comment (tolmachev21) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js]: Really needed, because I changed hard-code above using faker. And variable should truncate before testing

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js [hunk 13]
```diff
@@ -55,12 +54,12 @@ function checkSendingPossibility (newsItem) {
  * @param {string} taskId
  * @returns {Promise<void>}
  */
-async function sendNotifications (context, newsItem, taskId) {
+async function sendNotifications (newsItem, taskId) {
```
Comment (Alllex202) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js:57]: Also update the JSDoc for this function

**Review by Alllex202 — APPROVED** [review]
LGTM!

**Review by toplenboren — APPROVED** [review]
